### DISCUSSION TITLE
ExtendedMesh instances not in MainScene class

### DISCUSSION THREAD

**Discussion by sparrowflyy** [enable3d/enable3d#295] [General]
Hi,

I am trying to use threejs with enable3d's Physics Extension. I have my game scene class where AmmoPhysics is initialising and several classes of game objects  with ExtendedMesh inside. 
When i try to call constructor of my game scene in callback of PhysicsLoader the ExtendedMesh field in game objects classes is not initialized. Constructors of game objects are called from game scene constructor. 

Is there any possibility to have ExtendedMesh instances not only in scene classes? Or i just need to initialize all physic stuff directly in callback function of PhysicLoader without appealing to another classes containg ExtendedMesh , e.t.c.?

**Comment by yandeu** [comment]
I don't quite understand your question. Maybe share some code?

**Reply by sparrowflyy** [reply]
```js
PhysicsLoader('../../ammo', () => new MyScene());
/////
class MyScene {
  constructor() {
  ...
  this.gameObject = new GameObject()
  ...
  }
}

class GameObject {
  mesh: ExtendedMesh
  constructor() {
  ...
  this.mesh = new ExtendedMesh(...)
  ....
  }
```
  
  In this case ExtendedMesh in GameObject is not initializing. this.mesh is just undefined

**Reply by yandeu** [reply]
It works when I test it:

```ts
PhysicsLoader('/lib', () => new MyScene())

class MyScene {
  gameObject: GameObject
  constructor() {
    this.gameObject = new GameObject()
    console.log(this.gameObject.mesh)
  }
}

class GameObject {
  mesh: ExtendedMesh
  constructor() {
    this.mesh = new ExtendedMesh()
    console.log(this.mesh)
  }
}
````
